### ISSUE TITLE
crash hang on debian unstable

### ISSUE THREAD

**Issue by omac777** [ArturKovacs/emulsion#10]
clone the repo v1.9
cargo build --release
./target/release/emulsion

Expected:
to behave within gnome desktop.

Actual Result:
It froze the entire desktop.  Mouse was not moving.  Window was not responding.  Could not click Activities menu.  Could not close window.  In order to recover to a well-behaving desktop, I needed to switch to a non-gui pseudo terminal and "killall -r emulsion".

**Comment by omac777** [comment]
I got better behaviour in debian unstable when I passed a file to emulsion:
```
cd /home/blah/Downloads/pics10Sep2019/
/home/blah/emulsion/target/release/emulsion ./20190905_084455.jpg
```
It actually displayed the image and I was able to cursor right/left to see next/previous images.

I did notice there is a delay when flipping between images.  Change request to display progress of loading the app and the particular set of images that it is loading...at least when there is some display of some progress and that the app is moving forward, I wouldn't consider it a hang.  When it hangs the entire desktop however, that is very bad behaviour and the terminal progress should indicate there was an error at least.  I have no idea what caused the hang when I didn't specify a directory or file to the emulsion cli.

**Comment by ArturKovacs** [comment]
I'm rather confused about this issue. It also seems to be two issues actually.

1, emulsion froze the entire desktop.
2, emulsion works but you'd like to see some indication that it's loading a new image.

Please don't add unrelated requests or bug reports to an issue. Here are the reasons why:

First of all it makes it really hard to discuss them as the discussion thread is linear meaning that whatever I write will come after the previous message no matter if they were about the same thing or not. If we were to try to discuss two things at once in such a structure we would find it very confusing and hard to follow.

Second, it makes it more difficult to cleary referr to a specific problem in a commit message or a pull request. It's because referring to the issue number would refer to two things at once.

If you'd like to add something thats not related to the original post, please open a new issue so that we can comfortably discuss both in their own space. If you think the original has lost it's relevance or you think it's solved then go ahed and close the issue.

For now I opened a new issue for the second item https://github.com/ArturKovacs/emulsion/issues/new

**Comment by ArturKovacs** [comment]
With that said, I'm going to address the original post here.

Plese paste the output of:
```
glxinfo | grep -i opengl
```
You can install glxinfo with `sudo apt-get install mesa-utils`

**Comment by omac777** [comment]
On 4/23/20 3:08 AM, Artúr Kovács wrote:
> With that said, I'm going to address the original post here.
> 
> Plese paste the output of:
> ```
> glxinfo | grep -i opengl
> ```
> You can install glxinfo with `sudo apt-get install mesa-utils`
> 

glxinfo | grep -i opengl
OpenGL vendor string: X.Org
OpenGL renderer string: AMD Radeon (TM) R9 380 Series (TONGA, DRM
3.36.0, 5.5.0-2-amd64, LLVM 10.0.0)
OpenGL core profile version string: 4.6 (Core Profile) Mesa 20.0.4
OpenGL core profile shading language version string: 4.60
OpenGL core profile context flags: (none)
OpenGL core profile profile mask: core profile
OpenGL core profile extensions:
OpenGL version string: 4.6 (Compatibility Profile) Mesa 20.0.4
OpenGL shading language version string: 4.60
OpenGL context flags: (none)
OpenGL profile mask: compatibility profile
OpenGL extensions:
OpenGL ES profile version string: OpenGL ES 3.2 Mesa 20.0.4
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 3.20
OpenGL ES profile extensions:

**Comment by omac777** [comment]
On 4/23/20 3:08 AM, Artúr Kovács wrote:
> With that said, I'm going to address the original post here.
> 
> Plese paste the output of:
> ```
> glxinfo | grep -i opengl
> ```
> You can install glxinfo with `sudo apt-get install mesa-utils`
> 

On debian unstable, I just did a

```
git pull
cargo build --release
./target/release/emulsion /home/blah/Pictures/image20180915_114707888.jpg
```

The output was:

```

Found latest version tag v1.9
Current version is '1.10.0'
Parsed latest version is '1.9.0'

```

Most importantly however, when I cursor-right/left scrub or mouse-scrub
on the timeline cursor, it loads and displays the next images instantly
and without crashing anywhere.  Great improvement!  Hats off!

I'll come back to give you the same results in archlinux.

**Comment by ArturKovacs** [comment]
There was a bug that caused emulsion to hang right after startup on wayland desktops. The latest master should contain the fix. Could you check the latest master and see if that works in the environment where it previously froze the desktop?

**Comment by ArturKovacs** [comment]
I'm closing this as the issue (emulsion froze the entire desktop) is fixed to the best of my knowledge.
